code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -430,16 +430,21 @@
 			m_type = 1
 
 		if("giggle", "giggles")
+			var/M = handle_emote_param(param)
 			if(miming)
-				message = "<B>[src]</B> giggles silently!"
+				message = "<B>[src]</B> acts out a laugh[M ? " at [M]" : ""]."
 				m_type = 1
 			else
 				if(!muzzled)
-					message = "<B>[src]</B> giggles."
+					message = "<B>[src]</B> giggles[M ? " at [M]" : ""]."
 					m_type = 2
 				else
-					message = "<B>[src]</B> makes a noise."
-					m_type = 2
+					if(gender == FEMALE)
+						playsound(loc, 'sound/effects/womanlaugh.ogg', 50, 1)
+					else if(prob(50))
+						playsound(loc, 'sound/effects/manlaugh1.ogg', 50, 1)
+					else
+						playsound(loc, 'sound/effects/manlaugh2.ogg', 50, 1)
```

Comment:
This probably shouldn't play the same sound; giggling has a distinctively different style/flavor/persona about it than out and out laughing or chuckling does.